### PR TITLE
Add GEOADD and GEORADIUS commands

### PR DESCRIPTION
**Pull request by sklinkert**
This PR adds basic support for redis Geo commands GEOADD and GEORADIUS. I've also added a very simple test.

Credits: Thank you @alicebob for your help :-)

### CODE REVIEW

**Comment by alicebob** [comment]
> This PR adds basic support for redis Geo commands GEOADD and GEORADIUS. I've also added a very simple test.

Thanks! I'll have a look in the next few days.

File: cmd_geo.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+					Longitude: longitude,
+					Latitude:  latitude,
+				})
+			}
+		}
+
+		c.WriteLen(len(membersWithinRadius))
+		for _, member := range membersWithinRadius {
+			if withDist {
+				c.WriteLen(3)
```
Comment (alicebob) [inline cmd_geo.go]: That is not 3 if withCoord is false, so it really needs some tests :(

**Comment by alicebob** [comment]
Thanks,

the PR looks nice and simple. It does need more tests though:
- it needs some integration tests comparing output against a real redis. They are in the `./integration/` directory. See what's there for some inspiration.
- the unittests need more coverage. The integration tests can catch a lot of things, but things like invalid arguments, calling GEORADIUS on a string key do need some basic unit tests.
- I've changed the test to use the same redis client as all the other tests (github.com/gomodule/redigo). It's a bit more verbose, but maybe that's not so bad since it's more explicit in what it tests. Turns out I'm allowed to push to your branch :)

I'm happy to help with any or all of the above,
thanks!

**Comment by sklinkert** [comment]
> Thanks,
> 
> the PR looks nice and simple. It does need more tests though:
> 
> * it needs some integration tests comparing output against a real redis. They are in the `./integration/` directory. See what's there for some inspiration.
> * the unittests need more coverage. The integration tests can catch a lot of things, but things like invalid arguments, calling GEORADIUS on a string key do need some basic unit tests.
> * I've changed the test to use the same redis client as all the other tests (github.com/gomodule/redigo). It's a bit more verbose, but maybe that's not so bad since it's more explicit in what it tests. Turns out I'm allowed to push to your branch :)
> 
> I'm happy to help with any or all of the above,
> thanks!

Thanks for the review :-) Would be cool if you (or someone else) could add the integration tests since my time is a bit limited, unfortunately. I'll try to increase the unit test coverage and fix the `withCoord = false` bug.

I've also detected another issue while playing around: The returned coordinates are sometimes a bit imprecise from the inserted ones:

```
input:    {41.890214 12.490037 test  3}
output: {41.8916487 12.4895693 test  3}]
```

I think it's a precision issue because of converting types for geohash. Maybe you have an idea to fix it.

**Comment by alicebob** [comment]
> Would be cool if you (or someone else) could add the integration tests since my time is a bit limited, unfortunately.


ok, no problem

> I'll try to increase the unit test coverage and fix the `withCoord = false` bug.

thanks!

> I've also detected another issue while playing around: The returned coordinates are sometimes a bit imprecise from the inserted ones:

I'll see how real redis does it :)

**Comment by sklinkert** [comment]
I've added some tests. Please check if something is missing.

**Comment by alicebob** [comment]
On Fri, Jul 12, 2019 at 12:45:29AM -0700, Simon Klinkert wrote:
> I've added some tests. Please check if something is missing.

Gottit, thanks!

**Comment by alicebob** [comment]
Hi,

I've picked it up in #85  (it's a bit easier for me to work in a local branch). The GEORADIUS command has a lot of options, which I did not manage to finish yet, will continue another day.

**Comment by alicebob** [comment]
thanks again! There are still some commands left, but those are for another day :)

**Pull request closed without merging**